numpy/random/src/distributions/distributions.c

```diff
@@ -901,7 +901,7 @@ RAND_INT_TYPE random_logseries(bitgen_t *bitgen_state, double p) {
   double q, r, U, V;
   RAND_INT_TYPE result;
 
-  r = log(1.0 - p);
+  r = npy_log1p(-p);
```

Comment:
Here is can matter. I thnk for p very small this could lead to a steam break for RandomState, and so this code would need to be copied toa new function (`legacy_`) which would be used in `RandomState` and then the fix could be applied.